d3-dag.d.ts

```diff
@@ -0,0 +1,781 @@
+            ): number }
+
+            export interface Operator<NodeDatum, LinkDatum> {
+                /** This constructs a layered representation of the DAG meant for visualization.
+                 * The algorithm is based off ideas presented in K. Sugiyama et al. [1979],
+                 * but described by S. Hong. The sugiyama layout can be configured with different
+                 * algorithms for different stages of the layout.
+                 * For each stage there should be adecuate choices for methods that balance speed
+                 * and quality for your desired layout,
+                 * but any function that meets the interface for that stage is valid. */
+                (dag: Node<NodeDatum, LinkDatum>): Layout<NodeDatum, LinkDatum>
+                /** sets sugiyama to debug. 
+                * If debug is true, dummy nodes will be given more human readable ids,
+                * but this can cause conflicts with poorly chosen ids, so it it disabled by default. */
+                debug(debug: boolean): this
+                /** returns the current debug value, which defaults to false. */
+                debug() : this
+                /** sets this sugiyama layout's size to the specified two-element array of numbers [width, height].
+                 * When size is set, the minimum coordinate of every node is 0, and the maximum x and y coordinates
+                 * are width and height respectively. If the DAG only has one node vertically or horizontally,
+                 * it will be centered. */
+                size(size: [number, number]): this
+                /** returns the current layout size, which defaults to [1, 1], and is null if nodeSize was set.*/
+                size(): null|[number, number]
+                /** sets this sugiyama layout's nodeSize to the specified two-element array of numbers
+                 * [nodeWidth, nodeHeight]. When nodeSize is set, the minimum coordinate of every node is 0,
+                 * parents and children are at least nodeHeight appart, and neighboring nodes in the same layer
+                 * are at least nodeWidth apart. If the DAG only has one node vertically or horizontally,
+                 * it will be centered.*/
+                nodeSize(nodeSize: null|[number,number]) : this
+                /* returns the current layout node size, which defaults to null, and will return null if size was set. */
+                nodeSize(): null|[number,number]
+                /** sets the layering accessor to the specified function.
+                 * A layering accessor takes a dag and assigns every node a layer attribute
+                 * from zero to the number of layers - 1. See Sugiyama Layering Acessors. */
+                layering(layering: 
+                    Layering.Simplex|
+                    Layering.CoffmanGraham|
+                    Layering.LongestPath|
+                    Layering.Topological

```

Comment:
For all functions like this, I think the input should just be an interface that is common among all of these. The goal with have each step be composable was that it was possible for people to write their own or extend an existing one, even if that's somewhat unlikely.